engine/table/src/main/java/io/deephaven/engine/table/impl/indexer/DataIndexer.java

```diff
@@ -155,16 +164,39 @@ public boolean hasDataIndex(@NotNull final Collection<ColumnSource<?>> keyColumn
      * @param keyColumnNames The key column for which to retrieve a DataIndex
      * @return The {@link DataIndex}, or {@code null} if one does not exist
      */
-    public static DataIndex getDataIndex(@NotNull Table table, final String... keyColumnNames) {
+    @Nullable
+    public static DataIndex getDataIndex(@NotNull final Table table, final String... keyColumnNames) {
         if (keyColumnNames.length == 0) {
             return null;
         }
-        table = table.coalesce();
-        final DataIndexer indexer = DataIndexer.existingOf(table.getRowSet());
+        final Table tableToUse = table.coalesce();
+        final DataIndexer indexer = DataIndexer.existingOf(tableToUse.getRowSet());
+        if (indexer == null) {
+            return null;
+        }
+        return indexer.getDataIndex(getColumnSources(tableToUse, keyColumnNames));
+    }
+
+    /**
+     * If {@code table} has a DataIndexer, return a {@link DataIndex} for the given key columns, or {@code null} if no
```

Comment:
I had to duplicate some methods in this file to accept lists instead of string arrays. Let me know if I should keep it like this or do a conversion from list->array in calling code or array->list here?

Reply:
I'm fine with this if you do `hasDataIndex`, also, and delete the `getColumnSources` overload that takes an array.